spring-cloud-gcp-pubsub/src/test/java/com/google/cloud/spring/pubsub/core/PubSubTemplateTests.java

```diff
@@ -132,15 +133,18 @@ void testPublish_Object() throws IOException {
     doAnswer(
             invocation -> {
               PubsubMessage message = invocation.getArgument(1);
-              assertThat(message.getData().toStringUtf8())
-                  .isEqualTo(
-                      "{\"@class\":"
-                          + "\"com.google.cloud.spring.pubsub.core.test.allowed.AllowedPayload\""
-                          + ",\"name\":\"allowed\",\"value\":12345}");
+              char[] expected = ("{\"@class\":"
+                      + "\"com.google.cloud.spring.pubsub.core.test.allowed.AllowedPayload\""
+                      + ",\"name\":\"allowed\",\"value\":12345}").toCharArray();
+              char[] actual = message.getData().toStringUtf8().toCharArray();
+              Arrays.sort(expected);
+              Arrays.sort(actual);
```

Comment:
Another way to have fixed this test could have been to check that the size of the two arrays are the same and then just check whether `actual` contains all of the elements that are in `expected`. Such a fix would add more lines of code, but would also be more efficient than your fix. 

In any case, the developers may not care about this difference though, so you can keep your current fix as is.